app/controllers/early_allocations_controller.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+class EarlyAllocationsController < PrisonsApplicationController
+  before_action :load_prisoner, except: [:new]
+
+  def new
+    @early_assignment = EarlyAllocation.new offender_id_from_url
+  end
+
+  def create
+    @early_assignment = EarlyAllocation.new early_allocation_params.merge(offender_id_from_url)
+    if @early_assignment.stage1_complete?
+      stage2
+    else
+      stage1
+    end
+  end
+
+  def discretionary
+    @early_assignment = EarlyAllocation.new early_allocation_params.merge(offender_id_from_url)
+    if @early_assignment.valid?
+      save_model_and_email
+      render
+    else
+      render 'why'
+    end
+  end
+
+  def show
+    @early_assignment = EarlyAllocation.find_by! offender_id_from_url
+
+    respond_to do |format|
+      format.pdf
+    end
+  end
+
+private
+
+  def load_prisoner
+    @offender = OffenderService.get_offender(params[:prisoner_id])
+    @allocation = AllocationVersion.find_by!(offender_id_from_url)
+    @pom = PrisonOffenderManagerService.get_pom(@prison, @allocation.primary_pom_nomis_id)
+  end
+
+  def save_model_and_email
+    @early_assignment.save!
+
+    PomMailer.send_early_allocation_email(email: @offender.ldu.email_address,
```

Comment:
Do we have a clear steer on what is expected if there is no LDU email address, or do we not even get this far?

Reply:
Yes the story now includes a dead-end page if we don't have an LDU email address